godtools/ViewControllers/Tract/TractViewControllerPageMovements.swift

```diff
@@ -60,28 +77,79 @@ extension TractViewController {
         }
     }
     
-    fileprivate func moveViews() {
-        let newCurrentProgressViewFrame = CGRect(x: 0.0,
-                                                 y: 0.0,
-                                                 width: currentProgressWidth(),
-                                                 height: self.currentProgressView.frame.size.height)
+    // MARK: - Promises
+    
+    fileprivate func moveViews() -> Promise<Bool> {
+        let newCurrentProgressViewFrame = buildProgressViewFrame()
+        return UIView.promise(animateWithDuration: 0.35, delay: 0.0, options: .curveEaseInOut, animations: { 
+            self.currentProgressView.frame = newCurrentProgressViewFrame
+            for view in self.pagesViews {
+                view?.transform = CGAffineTransform(translationX: self.currentMovement, y: 0.0)
+            }
+        }).then { _ in
+            self.notifyCurrentViewDidAppearOnTract()
+            return Promise(value: true)
+        }
+    }
+    
+    fileprivate func moveForewards() -> Promise<Bool> {
+        self.currentPage += 1
+        let newCurrentProgressViewFrame = buildProgressViewFrame()
+        guard let currentPageView = self.view.viewWithTag(self.currentPage + self.viewTagOrigin) else {
+            return Promise(value: false)
+        }
         
-        UIView.animate(withDuration: 0.35,
-                       delay: 0.0,
-                       options: UIViewAnimationOptions.curveEaseInOut,
-                       animations: {
-                        self.currentProgressView.frame = newCurrentProgressViewFrame
-                        for view in self.pagesViews {
-                            view?.transform = CGAffineTransform(translationX: self.currentMovement, y: 0.0)
-                        } },
-                       completion: { finished in
-                        self.notifyCurrentViewDidAppearOnTract()
-        } )
+        return UIView.promise(animateWithDuration: 0.35, delay: 0.0, options: .curveEaseInOut, animations: {
+            self.currentProgressView.frame = newCurrentProgressViewFrame
+            currentPageView.transform = CGAffineTransform(translationX: self.currentMovement, y: 0.0)
+        }).then { _ in
+            self.moveViewsExceptCurrentViews(pageViews: [currentPageView])
+            return Promise(value: true)
+        }
     }
     
-    func notifyCurrentViewDidAppearOnTract() {
-        let currentTractView = self.view.viewWithTag(self.currentPage + self.viewTagOrigin) as! TractView
-        currentTractView.contentView?.notifyViewDidAppearOnTract()
+    fileprivate func moveBackwards() -> Promise<Bool> {
```

Comment:
This method and `moveForewards()` seem very similar. is there any way to reduce some of the duplicated logic?